test/common/common/block_memory_hash_set_test.cc

```diff
@@ -19,12 +22,15 @@ class BlockMemoryHashSetTest : public testing::Test {
   // TestValue that doesn't define a hash.
   struct TestValueBase {
     absl::string_view key() const { return name; }
-    void initialize(absl::string_view key) {
-      uint64_t xfer = std::min(sizeof(name) - 1, key.size());
+    void truncateAndInit(absl::string_view key, const Stats::StatsOptions& stats_options) {
+      uint64_t xfer = std::min(stats_options.maxNameLength(), key.size());
       memcpy(name, key.data(), xfer);
       name[xfer] = '\0';
     }
-    static uint64_t size() { return sizeof(TestValue); }
+    static uint64_t sizeGivenStatsOptions(const Stats::StatsOptions& stats_options) {
+      (void)stats_options;
```

Comment:
nit: `UNREFERENCED_PARAMATER` or just delete arg variable name. Same elsewhere if I missed it.